pkg/telemetryquerylanguage/functions/tqlotel/func_replace_all_patterns.go

```diff
@@ -23,7 +23,12 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
 )
 
-func ReplaceAllPatterns(target tql.GetSetter, regexPattern string, replacement string) (tql.ExprFunc, error) {
+const (
+	modeKey   = "key"
+	modeValue = "value"
+)
+
+func ReplaceAllPatterns(target tql.GetSetter, mode string, regexPattern string, replacement string) (tql.ExprFunc, error) {
```

Comment:
Can you update the telemetryquerylanguage README entry for this function to detail the new capabilities and the valid values for `mode`?

Reply:
Done, but I'm really not good at doc, please help optimize the description. =D